### ISSUE TITLE
"var _ ..." "For debugging; delete when done."

### ISSUE THREAD

**Issue by fgergo** [golang/go#15767]
1. What version of Go are you using (`go version`)?
b3bf2e7803 (latest commit at the moment)

3. What did you do?
cd src
grep -r "^var _ "|grep -v test

4. What did you expect to see?
no output, except for newish code

5. What did you see instead?
cmd/compile/internal/ssa/rewritedec.go:var _ = math.MinInt8 // in case not otherwise used
cmd/compile/internal/ssa/rewriteARM.go:var _ = math.MinInt8 // in case not otherwise used
cmd/compile/internal/ssa/rewriteAMD64.go:var _ = math.MinInt8 // in case not otherwise used
cmd/compile/internal/ssa/rewritegeneric.go:var _ = math.MinInt8 // in case not otherwise used
internal/syscall/windows/registry/zsyscall_windows.go:var _ unsafe.Pointer
internal/syscall/windows/zsyscall_windows.go:var _ unsafe.Pointer
io/multi.go:var _ stringWriter = (*multiWriter)(nil)
io/ioutil/ioutil.go:var _ io.ReaderFrom = devNull(0)
crypto/aes/aes_gcm.go:var _ gcmAble = (*aesCipherGCM)(nil)
crypto/aes/ctr_s390x.go:var _ ctrAble = (*aesCipherAsm)(nil)
crypto/aes/cbc_s390x.go:var _ cbcEncAble = (*aesCipherAsm)(nil)
crypto/aes/cbc_s390x.go:var _ cbcDecAble = (*aesCipherAsm)(nil)
net/http/server.go:var _ closeWriter = (*net.TCPConn)(nil)
net/http/server.go:var _ io.WriterTo = eofReader
syscall/mksyscall_windows.go:var _ unsafe.Pointer
syscall/zsyscall_windows.go:var _ unsafe.Pointer
encoding/json/stream.go:var _ Marshaler = (*RawMessage)(nil)
encoding/json/stream.go:var _ Unmarshaler = (*RawMessage)(nil)
database/sql/driver/types.go:var _ ValueConverter = boolType{}
database/sql/driver/types.go:var _ ValueConverter = int32Type{}
database/sql/driver/types.go:var _ ValueConverter = defaultConverter{}
database/sql/driver/driver.go:var _ Result = RowsAffected(0)
database/sql/driver/driver.go:var _ Result = noRows{}

**Comment by nightlyone** [comment]
TL;DR let's close it, this is just a misunderstanding.

Detailed explanations below:

> cmd/compile/internal/ssa/rewritedec.go:var _ = math.MinInt8 // in case not otherwise used
> cmd/compile/internal/ssa/rewriteARM.go:var _ = math.MinInt8 // in case not otherwise used
> cmd/compile/internal/ssa/rewriteAMD64.go:var _ = math.MinInt8 // in case not otherwise used
> cmd/compile/internal/ssa/rewritegeneric.go:var _ = math.MinInt8 // in case not otherwise used

This is generated code. Real sources are at cmd/compile/internal/ssa/gen/*.rules

> internal/syscall/windows/registry/zsyscall_windows.go:var _ unsafe.Pointer
> internal/syscall/windows/zsyscall_windows.go:var _ unsafe.Pointer

This is also generated code, generated by syscall/mksyscall_windows.go AFAICS.

> syscall/zsyscall_windows.go:var _ unsafe.Pointer

Generated.

> io/multi.go:var _ stringWriter = (multiWriter)(nil)
> io/ioutil/ioutil.go:var _ io.ReaderFrom = devNull(0)
> crypto/aes/aes_gcm.go:var _ gcmAble = (aesCipherGCM)(nil)
> crypto/aes/ctr_s390x.go:var _ ctrAble = (aesCipherAsm)(nil)
> crypto/aes/cbc_s390x.go:var _ cbcEncAble = (aesCipherAsm)(nil)
> crypto/aes/cbc_s390x.go:var _ cbcDecAble = (aesCipherAsm)(nil)
> net/http/server.go:var _ closeWriter = (net.TCPConn)(nil)
> net/http/server.go:var _ io.WriterTo = eofReader
> encoding/json/stream.go:var _ Marshaler = (RawMessage)(nil)
> encoding/json/stream.go:var _ Unmarshaler = (RawMessage)(nil)
> database/sql/driver/types.go:var _ ValueConverter = boolType{}
> database/sql/driver/types.go:var _ ValueConverter = int32Type{}
> database/sql/driver/types.go:var _ ValueConverter = defaultConverter{}
> database/sql/driver/driver.go:var _ Result = RowsAffected(0)
> database/sql/driver/driver.go:var _ Result = noRows{}

All these are compile time checks, that a type satisfies an interface. A common pattern in the Go world.
